samples/utils.py

```diff
@@ -0,0 +1,83 @@
+from test import constants
+# [START get_service]
+import uritemplate
+import httplib2
+import time
+import json
+import os
+
+from oauth2client.client import GoogleCredentials
+from googleapiclient.discovery import build_from_document
+# Libraries used by or included with Google APIs Client Library for Python
+from apiclient.discovery import DISCOVERY_URI
+from apiclient.errors import HttpError
+from apiclient.errors import InvalidJsonError
+
+
+def get_discovery_doc(api, version):
+    path = "".join([constants.DISCOVERY_DOC_PATH, api, version])
+    try:
+        age = time.time() - os.path.getmtime(path)
+        if age > constants.DISCOVERY_DOC_MAX_AGE:
+            update_discovery_doc(api, version, path)
+    except os.error:
+        update_discovery_doc(api, version, path)
+
+    with open(path, 'r') as discovery_doc:
+        return discovery_doc.read()
+
+
+def update_discovery_doc(api, version, path):
+    requested_url = uritemplate.expand(DISCOVERY_URI,
+                                       {'api': api, 'apiVersion': version})
+    resp, content = httplib2.Http().request(requested_url)
+    if resp.status >= 400:
+        raise HttpError(resp, content, uri=requested_url)
+    try:
+        with open(path, 'w') as discovery_doc:
+            discovery_json = json.loads(content)
+            json.dump(discovery_json, discovery_doc)
+    except ValueError:
+        raise InvalidJsonError(
+                'Bad JSON: %s from %s.' % (content, requested_url))
+
+
+def get_service():
+    credentials = GoogleCredentials.get_application_default()
```

Comment:
I think you tested this on GCE.

Did it work on your workstation too? I thought you need to call credentials.create_scoped(scopes) or something.